cookbook/entities/managing_tasks.rst

```diff
@@ -0,0 +1,254 @@
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+    use Symfony\Component\HttpFoundation\Request;
+
+    /**
+     * @Route("/task")
+     */
+    class TaskController extends Controller
+    {
+        /**
+         * @Route("/create", name="app_task_create")
+         * @Template("AppBundle:Task:update.html.twig")
+         */
+        public function createAction(Request $request)
+        {
+            return $this->update(new Task(), $request);
+        }
+
+        /**
+         * @Route("/edit/{id}", name="app_task_update", requirements={"id"="\d+"})
+         * @Template("AppBundle:Task:update.html.twig")
+         */
+        public function editAction(Task $task, Request $request)
+        {
+            return $this->update($task, $request);
+        }
+
+        private function update(Task $task, Request $request)
+        {
+            $form = $this->createForm(new TaskType(), $task);
+
+            return array(
+                'entity' => $task,
+                'form' => $form->createView(),
+            );
+        }
+    }
+
+.. _cookbook-entity-template:
+
+The Template
+------------
+
+The template that is responsible to display the form fields should extend the base template
+``OroUIBundle:actions:html.twig`` from the OroUIBundle. This templates defines some basic blocks
+that you can use. This way your own forms will provide the same look and feel as the ones coming
+with the OroPlatform:
+
+.. code-block:: html+jinja
+    :linenos:
+
+    {# src/AppBundle/Resources/views/Task/update.html.twig #}
+
+    {# extend the base template from the OroUIBundle #}
+    {% extends 'OroUIBundle:actions:update.html.twig' %}
+
+    {# reuse the form theme provided with the OroPlatform #}
+    {% form_theme form with 'OroFormBundle:Form:fields.html.twig' %}
+
+    {# make the current task accessible with the task variable #}
+    {% set task = form.vars.value %}
+
+    {# choose the appropriate action depending on whether a task is created or modified #}
+    {# this variable needs to be named formAction as this is what the base template expects #}
+    {% if task.id %}
+        {% set formAction = path('app_task_update', { 'id': task.id }) %}
+    {% else %}
+        {% set formAction = path('app_task_create') %}
+    {% endif %}
+
+    {% block navButtons %}
+        {# the cancelButton() macro creates a button that discards the
+           entered data and leads the user to the linked controller #}
+        {{ UI.cancelButton(path('app_task_index')) }}
+
+        {# the dropdownSaveButton() macro offers a way to let the user select
+           between different options when saving an entity, the selected option
+           will be passed to the controller handling the request as an additonal
+           parameter #}
+        {{ UI.dropdownSaveButton({
+            'html': UI.saveAndCloseButton() ~ UI.saveAndStayButton()
```

Comment:
There is a case when after creating the entity, user will not be able to edit the created record. We should check this case, so the right code will be:

```
{% set html = UI.saveAndCloseButton() %}
{% if task.id or resource_granted('app_task_update') %}
     {% set html = html ~ UI.saveAndStayButton() %}
 {% endif %}
 {{ UI.dropdownSaveButton({'html': html}) }}
```